### PR TITLE
Implement node:crypto getCipherInfo(...)

### PR DESCRIPTION
**Pull request by jasnell**

### CODE REVIEW

File: src/workerd/api/node/crypto.c++ [hunk 1]
```diff
@@ -1066,6 +1066,79 @@ jsg::BufferSource CryptoImpl::privateEncrypt(jsg::Lock& js,
...
 }
 
+namespace {
+ncrypto::Cipher getCipher(kj::OneOf<kj::String, int>& nameOrNid) {
+  KJ_SWITCH_ONEOF(nameOrNid) {
+    KJ_CASE_ONEOF(nid, int) {
+      return ncrypto::Cipher::FromNid(nid);
+    }
+    KJ_CASE_ONEOF(name, kj::String) {
+      std::string_view nameStr(name.begin(), name.size());
```
Comment (anonrig) [inline src/workerd/api/node/crypto.c++:1076]: Shouldn't this be cStr?

```suggestion
      std::string_view nameStr(name.cStr(), name.size());
```

File: src/workerd/api/node/crypto.c++ [hunk 2]
```diff
@@ -1066,6 +1066,79 @@ jsg::BufferSource CryptoImpl::privateEncrypt(jsg::Lock& js,
...
+            if (len != ivLength) return kj::none;
+            break;
+        }
+        ivLength = len;
+      }
+    }
+
+    auto nameView = cipher.getName();
+    auto modeView = cipher.getModeLabel();
+    kj::String name = kj::str(kj::heapArray<char>(nameView.data(), nameView.size()));
```
Comment (anonrig) [inline src/workerd/api/node/crypto.c++:1127]: At this point I feel like we should add a static method to kj string to initialize from string_view

File: src/workerd/api/node/crypto.h [hunk 3]
```diff
@@ -385,6 +385,26 @@ class CryptoImpl final: public jsg::Object {
       jsg::BufferSource buffer,
       PublicPrivateCipherOptions options);
 
+  struct CipherInfo {
+    kj::String name;
+    int nid;
+    jsg::Optional<int> blockSize;
+    jsg::Optional<int> ivLength;
+    int keyLength;
+    kj::String mode;  // 'cbc', 'ccm', 'cfb', 'ctr', 'ecb', 'gcm', 'ocb',
```
Comment (anonrig) [inline src/workerd/api/node/crypto.h:394]: Maybe a todo to make this an enum to avoid serialization?

File: src/workerd/api/node/crypto.h [hunk 4]
```diff
@@ -385,6 +385,26 @@ class CryptoImpl final: public jsg::Object {
...
+                      // 'ofb', 'stream', 'wrap', 'xts'
+    JSG_STRUCT(name, nid, blockSize, ivLength, keyLength, mode)
+  };
+
+  struct GetCipherInfoOptions {
+    jsg::Optional<int> keyLength;
```
Comment (anonrig) [inline src/workerd/api/node/crypto.h:400]: Should these be unsigned?

File: src/workerd/api/node/tests/crypto_cipher-test.js [hunk 5]
```diff
@@ -299,3 +301,47 @@ export const missingArgChecks = {
...
+      strictEqual(typeof info.keyLength, 'number');
+      strictEqual(typeof info.mode, 'string');
+    });
+  },
+};
+
+export const test_cipher_info = {
+  test() {
+    const info = getCipherInfo('aes-128-cbc', {
+      ivLength: 16,
```
Comment (anonrig) [inline src/workerd/api/node/tests/crypto_cipher-test.js:325]: Can you add a test that validates negative ivLength?

File: src/workerd/api/node/crypto.h (hunk 3 above)
Comment (jasnell) [reply src/workerd/api/node/crypto.h:394]: eh, this isn't that much of a hot path method in any use I've ever seen it in. Not convinced it is worth the added complexity of an enum.

File: src/workerd/api/node/crypto.h (hunk 4 above)
Comment (jasnell) [reply src/workerd/api/node/crypto.h:400]: They are ints in openssl and node.js.

File: src/workerd/api/node/crypto.c++ (hunk 2 above)
Comment (jasnell) [reply src/workerd/api/node/crypto.c++:1127]: Having a `KJ_STRINGIFY(std::string_view&)` somewhere should be enough. Can be done in a separate PR.

Comment (jasnell) [reply src/workerd/api/node/crypto.c++:1127]: Can be done with a KJ_STRINGIFY(...) function. Doesn't need to be in kj. Can be done in a separate PR.
